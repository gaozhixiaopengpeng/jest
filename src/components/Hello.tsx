import React, { CSSProperties, FC } from "react";
import { Col, Row } from "antd";

interface Props {
  type: "large" | "small";
  title: string;
}

const largeStyle: CSSProperties = {
  fontSize: "2rem",
  color: "red",
};

const smallStyle: CSSProperties = {
  fontSize: "1rem",
  color: "green",
};

const styles: Record<"small" | "large", CSSProperties> = {
  small: smallStyle,
  large: largeStyle,
};

const Hello: FC<Props> = (props) => {
  const { title, type } = { ...props };
  return (
    <Row style={styles[type]}>
      <Col>第一个 Col</Col>
      <Col>
        <div>{title}</div>
      </Col>
    </Row>
  );
};

export default Hello;
