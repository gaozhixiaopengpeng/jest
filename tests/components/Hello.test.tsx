import React from "react";
import { render } from "@testing-library/react";
import Hello from "@/components/Hello";

describe("Hello", () => {
  it("渲染小字体", () => {
    const { getByText } = render(<Hello type="small" title="Hello Jest" />);
    const content = getByText("Hello Jest");
    expect(content).toMatchSnapshot();
  });

  it("渲染大字体", () => {
    const { getByText } = render(<Hello type="large" title="Hello React" />);
    const content = getByText("Hello React");
    expect(content).toMatchSnapshot();
  });
});

