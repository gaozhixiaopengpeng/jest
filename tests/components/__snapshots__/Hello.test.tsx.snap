// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello 渲染大字体 1`] = `
<div>
  Hello React
</div>
`;

exports[`Hello 渲染小字体 1`] = `
<div>
  Hello Jest
</div>
`;
